src/backend/x11.rs

```diff
@@ -0,0 +1,182 @@
+//! Backend implementation for X11.
+use image::{Rgb, RgbImage};
+use x11rb::{
+    connection::Connection,
+    protocol::xproto::{ConnectionExt, ImageFormat, ImageOrder},
+};
+
+const ALL_BITS_FROM_PLANE: u32 = u32::MAX;
+const U8_MAX: f32 = u8::MAX as f32;
+
+/// A general enum with possible errors as values which can occur while
+/// operating with the xorg-server.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Couldn't connect to the xorg server: {0}")]
+    ConnectError(#[from] x11rb::errors::ConnectError),
+
+    #[error("The connection broke with the xorg server: {0}")]
+    ConnectionError(#[from] x11rb::errors::ConnectionError),
+
+    #[error("Couldn't request an image from the xorg-server: {0}")]
+    ReplyError(#[from] x11rb::errors::ReplyError),
+}
+
+/// The main function of this module.
+///
+/// This function collects, from each screen (a.k.a your monitors) a screenshot
+/// and returns it.
+pub fn get_images() -> Result<Vec<image::RgbImage>, Error> {
+    let (conn, _) = x11rb::connect(None)?;
+    let setup = conn.setup();
+
+    let mut images = Vec::with_capacity(setup.roots.len());
+
+    for screen in &setup.roots {
+        let width = screen.width_in_pixels as u32;
+        let height = screen.height_in_pixels as u32;
+
+        let (image_bytes, pixmap_format) = {
+            let cookie = conn
```

Comment:
cookie?

Reply:
> cookie?

yes, [that's how it's called](https://docs.rs/x11rb/0.12.0/x11rb/protocol/xproto/fn.get_image.html)